build.gradle.kts

```diff
@@ -103,14 +103,10 @@ tasks.addRule("Pattern: testGradle<ID>") {
         useJUnit {
             includeCategories("org.jenkinsci.gradle.plugins.jpi.UsesGradleTestKit")
         }
-        onlyIf {
-            gradleVersion.startsWith('4') && System.getProperty("java.specification.version") == "1.8" ||
-                    gradleVersion.startsWith('5')
-        }
     }
 }
 
-setOf("4.10.3", "5.6.4", "6.0.1")
```

Comment:
If possible I'd prefer to leave this because consumers tend to be [slow to upgrade](https://github.com/search?l=&p=1&q=distributionUrl+user%3Ajenkinsci+filename%3Agradle-wrapper.properties&type=Code). I would imagine a nice upgrade experience of a consumer on Gradle 4.x:

1. Bump jpi plugin to latest
2. Upgrade to next Gradle minor
3. Fix deprecation warnings
4. Commit
5. Repeat 2-5 for every newer version of Gradle

The nice thing about this is upgrades become purely about what changed in Gradle, rather than having to figure out what changed in Gradle and examine a compatibility matrix for the plugin.

I think this should be possible, and keep the code relatively clean, by creating a LegacyJpiPlugin that is applied if < Gradle 6. Open to feedback on this, but would like to explore at least.

Reply:
Moving to Gradle 6.0 as the minimum is OK, discussed further here: https://github.com/jenkinsci/gradle-jpi-plugin/pull/132#issuecomment-578829027